acceptable/openapi.py

```diff
@@ -22,32 +23,147 @@ def _to_dict(source: Any):
         return source
 
 
+def tidy_string(untidy: str):
+    tidy = str(untidy).replace("\n", " ")
+    while "  " in tidy:
+        tidy = tidy.replace("  ", " ")
+    return tidy.strip()
+
+
+def get_head_of_single_item_list(item_list, item_name):
+    count = len(item_list)
+    if count == 1:
+        return item_list[0]
+    else:
+        logging.warning(
+            f"Encountered {item_name} list with {count} items. Treating as empty."
+        )
+        return None
+
+
+def convert_endpoint_to_operation(endpoint: AcceptableAPI):
+    return OasOperation(
+        tags=[endpoint.service.group] if endpoint.service.group else ["none"],
+        summary=endpoint.title,
+        description=tidy_string(endpoint.docs),
+        operation_id=endpoint.name,
+    )
+
+
+@dataclass
+class OasOperation:
```

Comment:
Maybe move all types to the top of the file?

Reply:
Done.